### ISSUE TITLE
Swapchain resize on d3d12 fails

### ISSUE THREAD

**Issue by jdswebb** [turanszkij/WickedEngine#209]
Fails with:

```
D3D12 ERROR: ID3D12Resource2::<final-release>: CORRUPTION: An ID3D12Resource object (0x000002FF41239E40:'Unnamed Object') is referenced by GPU operations in-flight on Command Queue (0x000002FF3A763A60:'Unnamed ID3D12CommandQueue Object').  It is not safe to final-release objects that may have GPU operations pending.  This can result in application instability. [ EXECUTION ERROR #921: OBJECT_DELETED_WHILE_STILL_IN_USE]
D3D12: **BREAK** enabled for the previous message, which was: [ ERROR EXECUTION #921: OBJECT_DELETED_WHILE_STILL_IN_USE ]
```

I tried adding a `WaitForGPU()` call before but it didn't help - perhaps that function isn't working?

**Comment by jdswebb** [comment]
I also get the same error upon closing the editor.

**Comment by jdswebb** [comment]
I was able to fix it by creating an additional fence, signalling the direct command queue and waiting for it in `WaitForGPU()` which I'm calling from `GraphicsDevice_DX12::SetResolution`.

I could PR this change but not sure if you have a better solution?

**Comment by turanszkij** [comment]
Please open the PR, thanks

**Comment by turanszkij** [comment]
I'm not sure yet what goes wrong there, but I think it makes sense to decouple this from the main fence and frame count...

**Comment by turanszkij** [comment]
Fixed in #211
